packages/recommend-react/src/__tests__/useFrequentlyBoughtTogether.test.ts

```diff
@@ -83,4 +81,102 @@ describe('useFrequentlyBoughtTogether', () => {
       expect(result.current.recommendations).toEqual([250]);
     });
   });
+
+  test('returns FBT recommendations from initialState', async () => {
+    const { recommendClient } = createMockedRecommendClient();
+
+    const { result } = renderHook(
+      () =>
+        useFrequentlyBoughtTogether({
+          indexName: 'test',
+          recommendClient,
+          threshold: 0,
+          objectIDs: ['testing'],
+          queryParameters: {
+            facetFilters: ['test'],
+          },
+          transformItems: (items) => items,
+          initialState,
+        }),
+      {
+        wrapper: StrictMode,
+      }
+    );
+
+    await waitFor(() => {
+      expect(result.current.recommendations).toEqual(
+        initialState.recommendations
+      );
+      expect(result.current.status).toBe('idle');
+    });
+  });
+
+  test("doesn't initially fetch the recommendations with initialState", () => {
+    const { recommendClient } = createMockedRecommendClient();
+
+    renderHook(
+      () =>
+        useFrequentlyBoughtTogether({
+          indexName: 'test',
+          recommendClient,
+          threshold: 0,
+          objectIDs: ['testing'],
+          queryParameters: {
+            facetFilters: ['test'],
+          },
+          transformItems: (items) => items,
+          initialState,
+        }),
+      {
+        wrapper: StrictMode,
+      }
+    );
+
+    expect(recommendClient.getFrequentlyBoughtTogether).toHaveBeenCalledTimes(
+      0
+    );
+  });
+
+  test('fetches recommendations when props change with initialState', () => {
+    const { recommendClient } = createMockedRecommendClient();
+
+    const { rerender } = renderHook(
+      ({ indexName }) =>
+        useFrequentlyBoughtTogether({
+          indexName,
+          recommendClient,
+          threshold: 0,
+          objectIDs: ['testing'],
+          queryParameters: {
+            facetFilters: ['test'],
+          },
+          transformItems: (items) => items,
+          initialState,
+        }),
+      {
+        wrapper: StrictMode,
+        initialProps: { indexName: 'test' },
+      }
+    );
+    expect(recommendClient.getFrequentlyBoughtTogether).toHaveBeenCalledTimes(
+      0
+    );
+
+    act(() => {
+      rerender({ indexName: 'test1' });
+    });
+
+    expect(recommendClient.getFrequentlyBoughtTogether).toHaveBeenCalledTimes(
+      1
+    );
+    expect(recommendClient.getFrequentlyBoughtTogether).toHaveBeenCalledWith([
+      {
+        indexName: 'test1',
+        maxRecommendations: undefined,
+        objectID: 'testing',
+        queryParameters: { facetFilters: ['test'] },
+        threshold: 0,
+      },
+    ]);
+  });
```

Comment:
We can make these tests more resilient by asserting just what this test is responsible for. The rest should already be tested elsewhere.

```suggestion
    expect(recommendClient.getFrequentlyBoughtTogether).toHaveBeenCalledWith([
      expect.objectContaining({
        indexName: 'test1',
      }),
    ]);
```

Reply:
✅ : https://github.com/algolia/recommend/pull/109/commits/55280c4a963e58e3a535634c95961ba4342c01c9